src/main/java/com/cloudbees/jenkins/plugins/amazonecs/ECSService.java

```diff
@@ -232,7 +232,15 @@ TaskDefinition registerTemplate(final ECSCloud cloud, final ECSTaskTemplate temp
             LOGGER.log(Level.INFO, "Match on execution role: {0}", new Object[] {templateMatchesExistingExecutionRole});
             LOGGER.log(Level.FINE, "Match on execution role: {0}; template={1}; last={2}", new Object[] {templateMatchesExistingExecutionRole, template.getExecutionRole(), currentTaskDefinition.getExecutionRoleArn()});
 
-            templateMatchesExistingNetworkMode = StringUtils.equals(StringUtils.defaultString(template.getNetworkMode()), StringUtils.defaultString(currentTaskDefinition.getNetworkMode()));
+            
+            if(!StringUtils.isEmpty(template.getNetworkMode())){
```

Comment:
Should this be `if(StringUtils...)` without the not?
The comment suggests the then branch is for the empty string but the condition checks for not the empty string.